File: java/darts/src/main/java/Darts.java
class Darts {

    private final int score;

    Darts(double x, double y) {
        score = calculate(x, y);
    }

    private int calculate(double x, double y) {
        if (Math.abs(x - y) < 0.01) return 10;
        if ((x <= 5) && (y <= 5)) return 5;
        if (x <= 10) return 1;
        return 0;
    }

    int score() {
        return score;
    }

}

File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    deltaX = 0 - x
    deltaY = 0 - y
    sum = (deltaX * deltaX) + (deltaY * deltaY)
    # sqrt of (0 - x)^2 + (o-y)^2
    distance = :math.sqrt(sum)
    cond do
      distance <= 5 -> 5
      distance <= 10 -> 1
      true -> 0
    end
  end
end
